Source/Core/Common/MemoryUtil.cpp

```diff
@@ -50,6 +54,35 @@ void* AllocateExecutableMemory(size_t size)
 
   return ptr;
 }
+// Certain platforms (Mac OS X on ARM) enforce that a single thread can only have write or
+// execute permissions to pages at any given point of time. The two below functions
+// are used to toggle between having write permissions or execute permissions.
+//
+// The default state of these allocations in Dolphin is for them to be executable,
+// but not writeable. So, functions that are updating these pages should wrap their
+// writes like below:
+
+// JITPageWriteEnableExecuteDisable();
+// PrepareInstructionStreamForJIT();
+// JITPageWriteDisableExecuteEnable();
+
+//Allows a thread to write to executable memory, but not execute the data.
+void JITPageWriteEnableExecuteDisable(){
```

Comment:
There is at least one place currently where this function-pair is nested. One should probably increment a nest count and either (a) only re-enable execution once it returns to zero, or (b) fire an assertion when this happens.

Reply:
Good idea, I will make it explicitly nestable in the next push.